### PR TITLE
Migrate Consul on ECS to use Consul Dataplane instead of client agents

### PR DESCRIPTION
**Pull request by Ganeshrockz**
## Changes proposed in this PR:
- Accumulates all the changes present in the feature branch for the new Dataplane based architecture.

## How I've tested this PR:

CI, Terraform modules

## How I expect reviewers to test this PR:

Almost all the files have already been reviewed in individual PRs raised towards the feature branch. I would need reviewers to look at the CHANGELOG because that's the only addition compared to the existing changes in the feature branch.

## Checklist:
- [X] Tests added
- [X] CHANGELOG entry added

    [HashiCorp engineers only. Community PRs should not add a changelog entry.]::
    [Changelog entries should use present tense, e.g. "Add support for..."]::

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,44 @@
+## Unreleased
+
+FEATURES
+* Removes the need for a Consul client when running Consul on ECS and transitions to the [Consul Dataplane](https://developer.hashicorp.com/consul/docs/connect/dataplane) based architecture.
```
Comment (Ganeshrockz) [inline CHANGELOG.md:4]: Should I add this PR's link to all these points?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,44 @@
...
+   - Add a new top level required field `consulServers` to help ecs commands connect to the Consul servers without the client agent. Holds details about the server address and protocol specific TLS settings.
+   - Remove the `service.checks` field. Without client agent, Consul will no longer be able to run user defined checks for their service instances.
+   - Add the `proxy.healthCheckPort` field which can be hit to determine Envoy's readiness.
+   - Add the `proxy.upstreams.destinationPeer` field to enable the proxy to hit upstreams present in peer Consul clusters.
+   - Add the `meshGateway.healthCheckPort` field which can be hit to determine Envoy's readiness.
+* Allow configuring TLS certs for gRPC traffic to Consul servers with `CONSUL_GRPC_CACERT_PEM` or `consulServers.defaults.caCertFile` or `consulServers.grpc.caCertFile` config option.
+* Allow configuring TLS certs for HTTP traffic to Consul servers with `CONSUL_HTTPS_CACERT_PEM` or `consulServers.defaults.caCertFile` or `consulServers.http.caCertFile` config option.
+* Add the [go-discover](https://github.com/hashicorp/go-discover) binary to the Consul ECS image to better support the discovery of Consul servers.
```
Comment (pglass) [inline CHANGELOG.md:37]:
```suggestion
* Add the [go-discover](https://github.com/hashicorp/go-discover) binary to the Consul ECS image to better support [cloud auto-join](https://developer.hashicorp.com/consul/docs/install/cloud-auto-join).
```

**Comment by Ganeshrockz** [comment]
@pglass Thanks for all the suggestions. Addressed all the comments

**Review by cthain — APPROVED** [review]
This is excellent @Ganeshrockz. Great job implementing this and shepherding all the PRs through. :clap:

**Pull request merged**